src/Build/BackEnd/Components/SdkResolution/NuGet/NuGetSdkResolver.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using NuGet.Commands;
+using NuGet.Configuration;
+using NuGet.LibraryModel;
+using NuGet.Packaging;
+using NuGet.Versioning;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using SdkResolverContextBase = Microsoft.Build.Framework.SdkResolverContext;
+using SdkResultBase = Microsoft.Build.Framework.SdkResult;
+using SdkResultFactoryBase = Microsoft.Build.Framework.SdkResultFactory;
+
+namespace Microsoft.Build.BackEnd.SdkResolution.NuGet
+{
+    /// <summary>
+    /// Represents a NuGet-based SDK resolver.  It is very important that this class does not reference any NuGet assemblies
+    /// directly as an optimization to avoid loading them unless they are needed.  The current implementation only loads
+    /// Newtonsoft.Json if a global.json is found and it contains the msbuild-sdks section and a few NuGet assemblies to parse
+    /// a version.  The remaining NuGet assemblies are then loaded to do a restore.
+    /// </summary>
+    internal sealed class NuGetSdkResolver : NuGetSdkResolverBase
+    {
+        public override string Name => nameof(NuGetSdkResolver);
+
+        public override int Priority => 2500;
+
+        public override SdkResultBase Resolve(SdkReference sdk, SdkResolverContextBase context, SdkResultFactoryBase factory)
+        {
+            object parsedSdkVersion;
```

Comment:
Why object?

Reply:
Ah, comment in the nested class :)

---
In reply to: [160777249](https://github.com/Microsoft/msbuild/pull/2850#discussion_r160777249) [](ancestors = 160777249)